### ISSUE TITLE
InvalidCastException: Unable to cast object for Enum.GetName()

### ISSUE THREAD

**Issue by jamesholcomb** [hazzik/DelegateDecompiler#67]
Hi, great library.  I ran into a scenario on 0.18 using `Enum.GetName()`.  Please see the following repro:
````
[Test]
public void Should_be_able_to_decompile_enum()
{
	var actual = Enumerable.Range(1, 3)
		.Select(x => new Foo { Value = x })
		.AsQueryable()
		.Select(x => x.EnumName)
		.Decompile();

	actual.First().Should().Be("A");
}

public enum FooEnum { A = 1, B = 2, C = 3 }
public class Foo
{
	public int Value { get; set; }

	[Computed]
	public string EnumName
	{
		get
		{
			return Enum.GetName(typeof(FooEnum), this.Value);
		}
	}
}
````
Result StackTrace:	
```
   at DelegateDecompiler.Processor.Process()
   at DelegateDecompiler.Processor.Process(VariableInfo[] locals, IList`1 args, Instruction instruction, Type returnType)
   at DelegateDecompiler.MethodBodyDecompiler.Decompile()
   at DelegateDecompiler.DecompileExtensions.<>c__DisplayClass1.<Decompile>b__0(MethodInfo m)
   at DelegateDecompiler.Cache`2.GetOrAdd(TKey key, Func`2 func)
   at DelegateDecompiler.DecompileExtensions.Decompile(MethodInfo method)
   at DelegateDecompiler.DecompileExpressionVisitor.Decompile(MethodInfo method, Expression instance, IList`1 arguments)
   at DelegateDecompiler.DecompileExpressionVisitor.VisitMember(MemberExpression node)
   at System.Linq.Expressions.MemberExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitLambda[T](Expression`1 node)
   at System.Linq.Expressions.Expression`1.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitUnary(UnaryExpression node)
   at System.Linq.Expressions.UnaryExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitArguments(IArgumentProvider nodes)
   at System.Linq.Expressions.ExpressionVisitor.VisitMethodCall(MethodCallExpression node)
   at DelegateDecompiler.DecompileExpressionVisitor.VisitMethodCall(MethodCallExpression node)
   at System.Linq.Expressions.MethodCallExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at DelegateDecompiler.DecompiledQueryProvider.CreateQuery[TElement](Expression expression)
   at DelegateDecompiler.DecompileExtensions.Decompile[T](IQueryable`1 self)
Result Message:	System.InvalidCastException : Unable to cast object of type 'System.RuntimeType' to type 'System.Reflection.FieldInfo'.
```

**Comment by hazzik** [comment]
Did it work previously?

**Comment by hazzik** [comment]
@jamesholcomb this has been fixed and released to nuget as 0.18.1
